### DISCUSSION TITLE
Use custom JLink Commander script

### DISCUSSION THREAD

**Discussion by jcpince** [zephyrproject-rtos/zephyr#79125] [RFC]
Hello,

I'm working on a new CPU port and I need to modify the JLink commander script used (the platform requires a reset using the reset pin before flashing).

I'd like to add an option to the JLink arguments in west flash to allow using a custom commander script and settings as follow:

```
board_runner_args(jlink` "--iface=SWD" "--speed=4000" "--commands=myscript" "--settings=mysettings")
```

**Comment by henrikbrixandersen** [comment]
You can pass whatever options supported by the tool using `--tool-opt=`.

**Reply by jcpince** [reply]
Hello Henri,
Thanks for the suggestion, that's actually the way I pass my settings file. However, the jlink.py script writes a commander script in /tmp and passes it as an argument to JLink as follows:

> /opt/SEGGER/JLink_V798e/JLinkExe -nogui 1 -if SWD -speed 4000 -CommanderScript /tmp/tmpsbau368wjlink/runner.jlink -nogui 1

So, we need to replace the built script with the one that could be platform specific when necessary and adding it to tool-opt is not enough...

**Reply by henrikbrixandersen** [reply]
I am using `--tool-opt=-jlinkscriptfile /path/to/out/of/tree/boards/common/myscript.JLinkScript` here. Have you tried that?

**Reply by jcpince** [reply]
The jlinkscriptfile is not intended to pass commands but to allow specific HW init, reset...

So, it's not helping here. If you pass the commander script, you'll get an error such as:
> ****** Error: Error while evaluating J-Link script file: Error while compiling. Line 1, column 1:
> ExitOnError 1
> ^
> Expected storage type
> Failed prepare script file

**Reply by henrikbrixandersen** [reply]
Ok, that's fair. Sounds like we need a new runner option, then. 

Can you draft a PR for adding this?

**Comment by jcpi-q** [comment]
Sorry it took a while, I had to create an account first :-P

https://github.com/zephyrproject-rtos/zephyr/pull/79372

**Reply by jcpi-q** [reply]
@henrikbrixandersen, I'm not sure of the process here. There are 4 reviewers assigned but not sure if I should ping them and leave it be until reviewed.
Any suggestion?
Thanks!
JC
